rasgotransforms/rasgotransforms/transforms/metric/metric.yaml

```diff
@@ -0,0 +1,56 @@
+name: metric
+type: insight
+tags:
+  - insight
+description: Calculate metric values given a metric definition
+arguments:
+  timestamp:
+    type: column
+    description: The date/datetime column by which the metric will be aggregated
+  target_sql:
+    type: column
+    description: The target column used to calculate the metric, can also be a sql string '(e.g. cast(target_column as string)')
+  aggregation_type:
+    type: string
+    description: the Aggregation method used to calculate the metric (e.g. 'sum', 'count_distinct', 'max')
+  time_grain:
+    type: string
+    description: The time grain (day, week, month, quarter, year) used to create the datespine that the metric is aggregated by
+  dimensions:
+    type: column_list
+    description: List of columns used to group by in metric aggregation
+    is_optional: true
+  filters:
+    type: list
```

Comment:
should we use `    type: filter_list` here so that the UI input will match? in case users want to edit a metric rendering step